solidity/contracts/system/TBTCSystem.sol

```diff
@@ -142,13 +144,61 @@ contract TBTCSystem is Ownable, ITBTCSystem, DepositLog {
             _keepThreshold,
             _keepSize
         );
+        tbtcToken = _tbtcToken;
         setTbtcDepositToken(_tbtcDepositToken);
-        _initialized = true;
+        initializedTimestamp = block.timestamp;
         allowNewDeposits = true;
     }
 
-    /// @notice gets whether new deposits are allowed.
-    function getAllowNewDeposits() external view returns (bool) { return allowNewDeposits; }
+    /// @notice returns whether new deposits should be allowed.
+    function getAllowNewDeposits() external view returns (bool) {
+        if (initializedTimestamp == 0 || !allowNewDeposits) {
```

Comment:
If `initializedTimestamp == 0` then `allowNewDeposits == false`, as the only write to it is in `initialize`, which sets `allowNewDeposits` to `true`.

Reply:
Right now :) I'd rather keep the semantic distinction unless you think the gas matters.

Reply:
No longer relevant